src/ui/state/settings.rs

```diff
@@ -0,0 +1,83 @@
+use std::convert::From;
+use std::sync::Arc;
+
+use iced::{Command, Element};
+
+use super::State;
+
+use crate::revaultd::{
+    config::{default_config_path, Config},
+    RevaultD,
+};
+use std::path::PathBuf;
+
+use crate::ui::{
+    error::Error,
+    message::Message,
+    view::{Context, SettingsView},
+};
+
+#[derive(Debug)]
+pub struct SettingsState {
+    revaultd: Arc<RevaultD>,
+    view: SettingsView,
+    // TODO: is this needed?
+    warning: Option<Error>,
+    config: Option<Config>,
+    revaultd_config_path: Option<PathBuf>,
+}
+
+impl SettingsState {
+    pub fn new(revaultd: Arc<RevaultD>, revaultd_config_path: Option<PathBuf>) -> Self {
```

Comment:
sorry I didnt warn you but revaultd embedded the config, I do not think we need to open it again.
```rust
pub fn new(config: revaultd::config::Config) -> Self {
```